### ISSUE TITLE
webpack 5.0.0-rc.1 regression --watch CLI option broke

### ISSUE THREAD

**Issue by EvHaus** [webpack/webpack#11553]
# Bug report

**What is the current behavior?**

Looks like there is a regression between rc.0 and rc.1.

Adding the `--watch` option to the `webpack` CLI stopped working in `rc.1` and now returns the following error:

```
/myproject/node_modules/webpack-cli/bin/cli.js:281
				throw err;
				^

Error: A 'callback' argument need to be provided when the 'watch' option is set
    at webpack (/myproject/node_modules/webpack/lib/webpack.js:140:10)
    at f (/myproject/node_modules/webpack/lib/index.js:31:15)
    at processOptions (/myproject/node_modules/webpack-cli/bin/cli.js:272:16)
    at /myproject/node_modules/webpack-cli/bin/cli.js:364:3
    at Object.parse (/myproject/node_modules/webpack-cli/node_modules/yargs/yargs.js:576:18)
    at /myproject/node_modules/webpack-cli/bin/cli.js:49:8
    at Object.<anonymous> (/myproject/node_modules/webpack-cli/bin/cli.js:366:3)
    at Module._compile (internal/modules/cjs/loader.js:1076:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at Module.require (internal/modules/cjs/loader.js:965:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/myproject/node_modules/webpack/bin/webpack.js:132:2)
    at Module._compile (internal/modules/cjs/loader.js:1076:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1097:10)
    at Module.load (internal/modules/cjs/loader.js:941:32)
    at Function.Module._load (internal/modules/cjs/loader.js:782:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47
```

Rolling back to rc.0 makes the error go away.

**If the current behavior is a bug, please provide the steps to reproduce.**

Using any `webpack.config.js` file run `webpack --watch` instead of the regular `webpack` command

**What is the expected behavior?**

`webpack --watch` should work without any errors as it does with `rc.0`.

**Other relevant information:**
webpack version: 5.0.0-rc.1
Node.js version:  14.10.1
Operating System: macOS
Additional tools: N/A

**Comment by evilebottnawi** [comment]
What is version of webpack-cli?

**Comment by EvHaus** [comment]
webpack-cli `3.3.12` and `4.0.0-rc.0` both produce the same error.

**Comment by evilebottnawi** [comment]
@EvHaus Thanks

**Comment by sokra** [comment]
Let's revert this change to a deprecation instead

**Comment by Krutsch** [comment]
I can see the same issue without having a config file. 
`webpack -w in.js -o out.js`
`[webpack-cli] Error: A 'callback' argument need to be provided when the 'watch' option is set`
`[webpack-cli] TypeError: Cannot read property 'compilers' of undefined`

**Comment by sokra** [comment]
Thanks for the report. Fix is on the way and I'll publish a new RC after that.

**Comment by sokra** [comment]
fix in 5.0.0-rc.2

**Comment by EvHaus** [comment]
Confirmed. 5.0.0-rc.2 fixes the problem for me.
